CRM/Member/BAO/Membership.php

```diff
@@ -88,17 +88,8 @@ public static function add(&$params) {
     //it is set during renewal of membership.
     $logStartDate = $params['log_start_date'] ?? NULL;
     $logStartDate = ($logStartDate) ? CRM_Utils_Date::isoToMysql($logStartDate) : CRM_Utils_Date::isoToMysql($membership->start_date);
-    $values = self::getStatusANDTypeValues($membership->id);
-
-    $membershipLog = [
-      'membership_id' => $membership->id,
-      'status_id' => $membership->status_id,
-      'start_date' => $logStartDate,
-      'end_date' => CRM_Utils_Date::isoToMysql($membership->end_date),
-      'modified_date' => CRM_Utils_Time::date('Ymd'),
-      'membership_type_id' => $values[$membership->id]['membership_type_id'],
-      'max_related' => $membership->max_related,
-    ];
+    $membershipTypeID = (int) self::getStatusANDTypeValues($membership->id)[$membership->id]['membership_type_id'];
+    self::createMembershipLog($membership, $logStartDate, $membershipTypeID);
 
     if (!empty($params['modified_id'])) {
       $membershipLog['modified_id'] = $params['modified_id'];
```

Comment:
Shouldn't this code be removed, up to and including the call to CRM_Member_BAO_MembershipLog::add below?

Reply:
@Sjord yes, thanks!